src/components/shared/Countdown.tsx

```diff
@@ -0,0 +1,63 @@
+import { StyleSheet, View, Text, StyleProp, TextStyle } from "react-native";
+import * as React from "react";
+
+interface CountdownProps {
+  time: number; // in seconds
+  counter: number; // in seconds
+  style?: StyleProp<TextStyle>;
+  onFinish: () => void;
+}
+
+interface CountdownState {
+  timeRemaining: number;
+}
+
+export default class Countdown extends React.PureComponent<
+  CountdownProps,
+  CountdownState
+> {
+  static defaultProps = {
+    time: 30,
+    counter: 1000
+  };
+
+  state = {
+    timeRemaining: this.props.time
```

Comment:
Shouldn't this be written in getDerivedStateFromProps?
What if props change?

Reply:
Yes and it should be renamed to `initialInterval` or `timerInterval`